account_bank_statement_import_online_nordigen/models/online_bank_statement_provider_nordigen.py

```diff
@@ -0,0 +1,465 @@
+
+    def _get_nordigen_institutions(self, country_code=""):
+        try:
+            url = NORDIGEN_ENDPOINT + "/institutions/"
+            response = requests.get(
+                url,
+                params={
+                    "country": country_code,
+                },
+                headers=self._get_nordigen_headers(self.nordigen_token),
+            )
+            if response.status_code == 200:
+                institutions = json.loads(response.text)
+                return institutions
+            else:
+                return []
+        except Exception as e:
+            _logger.debug("Getting Institutions  %s", e)
+            return []
+
+    def _check_bank_account_nordigen(self):
+        if not self.journal_id.bank_account_id:
+            raise UserError(
+                _("To continue configure bank account on journal %s")
+                % (self.journal_id.display_name)
+            )
+
+    def action_select_nordigen_bank(self):
+        country_model = self.env["res.country"]
+        self._check_bank_account_nordigen()
+        ctx = self.env.context.copy()
+        country = self.company_id.country_id
+        self._update_token_nordigen()
+        institutions = self._get_nordigen_institutions()
+        country_codes = []
+        for institution in institutions:
+            country_codes += institution["countries"]
+        country_codes = list(set(country_codes))
+        countries = country_model.search([("code", "in", country_codes)])
+        country_names = [{"code": c.code, "name": c.name} for c in countries]
+        ctx.update(
+            {
+                "dialog_size": "medium",
+                "country": country.code,
+                "country_name": country.name,
+                "journal_id": self.journal_id.id,
+                "institutions": institutions,
+                "country_names": country_names,
+            }
+        )
+        return {
+            "type": "ir.actions.client",
+            "tag": "online_sync_institution_selector_nordigen",
+            "name": _("Select Bank of your Account"),
+            "params": {},
+            "target": "new",
+            "context": ctx,
+        }
+
+    def _create_redirect_url_nordigen(self):
+        base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
+        return url_join(base_url, "nordigen/response")
+
+    def action_check_agreement(self):
+        self._update_token_nordigen()
+        institution_id = self.nordigen_institution_id
+        if (
+            self.env["ir.module.module"]
+            .sudo()
+            .search(
+                [("name", "=", "account_bank_statement_import_online_nordigen")],
+                limit=1,
+            )
+            .demo
+        ):
+            institution_id = SANDBOX_INSTITUTION_ID
+        self.nordigen_last_requisition_ref = str(uuid4())
+        # TO DELETE
+        institution_id = "SANDBOXFINANCE_SFIN0000"
```

Comment:
Only thing I see stopping a merge is this hardcode, either it should be removed or made into a UI based switch for demo

Reply:
Happy to approve after. (Been testing for 2 months)

Reply:
OK, so who should comment that line out? :-)

Reply:
@ayushin Normally the person who initially posted the PR.
@JordiBForgeFlow are you still wanting this PR or should it be forked to another PR

Reply:
@Entrepreneur-AJ feel free to propose the fix in another PR. Then we can close this one.

Reply:
Will look at sorting it after work tonight